flink/v1.16/flink/src/main/java/org/apache/iceberg/flink/FlinkCatalog.java

```diff
@@ -148,6 +150,17 @@ private Namespace toNamespace(String database) {
   }
 
   TableIdentifier toIdentifier(ObjectPath path) {
+    String objectName = path.getObjectName();
+    List<String> tableName = Splitter.on('$').splitToList(objectName);
+    if (tableName.size() > 1 && MetadataTableType.from(tableName.get(1)) != null) {
```

Comment:
Check whether the name is points to a metadata table.

Reply:
we can apply more strict check here. if `objectName` contains `$`, it should only be one occurrence. In another word, `tableName.size()` should be 2.

Reply:
Done